### PR TITLE
completions/make: Supporting GNUmakefile and makefile

### LINKED ISSUE #216: [make completion] Adding support for GNUmakefile and makefile
Hi,

The [make completion](https://github.com/ohmybash/oh-my-bash/blob/master/completions/makefile.completion.sh) plugin is looking only for the file "Makefile" but make also support GNUmakefile and makefile. [source](https://linux.die.net/man/1/make)

Current bebavior:
```bash
make grep: Makefile: No such file or directory
```

Note: the same error apply if no makefile are in the current directory.

### PR DESCRIPTION
**Pull request by jeansebastienh**
Close #216

### CODE REVIEW

File: completions/makefile.completion.sh
Comment (jeansebastienh) [reply completions/makefile.completion.sh]: Good suggestion :)

Thanks!

Comment (akinomyoga) [reply completions/makefile.completion.sh]: Thank you for the update! But I guess you have forgotten to change `-name` to `-iname`. If you use `-name`, `'*makefile'` doesn't match `Makefile`.

[ Note: If you care about the POSIX standard which doesn't define `-iname`, `-maxdepth` is also not defined in the POSIX standard. As far as I have tested, `-iname` was supported by `find` in all the systems (GNU/Linux, FreeBSD, Minix, Solaris, Haiku), but `-maxdepth` was not supported by Solaris `find`. ]

Comment (jeansebastienh) [reply completions/makefile.completion.sh]: It was a mistake. Sorry for that.

Hum good catch for the maxdepth. I don't know what are the requirement for this MR to be accepted.
In the readme it's :
> Unix-like operating system (macOS or Linux)

Comment (akinomyoga) [reply completions/makefile.completion.sh]: OK. I think you can forget about Solaris, which intentionally preserves very old behavior of UNIX. It doesn't even satisfy the latest version of POSIX by default. As README says macOS or Linux, I think it's enough to test it in macOS and GNU/Linux.

Comment (akinomyoga) [reply completions/makefile.completion.sh]: > As README says macOS or Linux, I think it's enough to test it in macOS and GNU/Linux (and BSDs as an option).

FYI, I got a reply from @nntoan about the compatibility in the reply https://github.com/ohmybash/oh-my-bash/discussions/268#discussioncomment-1862992.

> Just keep in mind we need all PRs to be POSIX-compatible

**Comment by akinomyoga** [comment]
As it turned out in https://github.com/ohmybash/oh-my-bash/discussions/268#discussioncomment-1862992 that `oh-my-bash` actually requires the POSIX compatibility, we need some adjustments here. One option is to filter the output of `find` using `grep` (but this might detect non-existent Makefile's when filenames contain newlines).

**Comment by jeansebastienh** [comment]
Hi,
I tried some solutions but didn't find anything working properly.

Could you propose a patch if you have an idea ?

**Comment by akinomyoga** [comment]
OK, I'll try to think about it. Thank you for your reply.

File: completions/makefile.completion.sh
Comment (akinomyoga) [inline completions/makefile.completion.sh:3]: @jeansebastienh I haven't tested it, but how about the following one?

```suggestion
complete -W "\`shopt -u nullglob; shopt -s nocaseglob; grep -oE '^[a-zA-Z0-9_-]+:([^=]|$)' *makefile 2>/dev/null | sed 's/[^a-zA-Z0-9_-]*$//'\`" make
```

Comment (jeansebastienh) [reply completions/makefile.completion.sh]: It's working like a charm. 

Thanks 👍🏼

Comment (akinomyoga) [reply completions/makefile.completion.sh]: Thank you for checking! LGTM now. Then let's merge this PR! After applying additional cosmetic changes, I'll merge it. (You might have already noticed it, but I joined the maintainer team last week so can process PRs now.)

**Review by akinomyoga — APPROVED** [review]

**Comment by akinomyoga** [comment]
I've merged it. Thank you for your quick responses to suggestions!

**Pull request merged**
